app/models/user.rb

```diff
@@ -7,4 +7,12 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  def can_edit_piece?(piece)
+    self.admin || self == piece.user
+  end
+
+  def can_edit_review?(review)
+    self.admin || self == review.user
```

Comment:
Another way to potentially refactor this might be:

```
def can_edit?(resource)
  admin || owns?(resource)
end

def owns?(resource)
  return false unless resource && resource.user
  self == resource.user
end
```